pkg/services/ngalert/sender/dispatcher.go

```diff
@@ -0,0 +1,280 @@
+package sender
```

Comment:
What's new in this file, and what is copied from other code? I want to know how closely to review code.

Reply:
Almost 100% copy/past